account_chart_update/wizard/wizard_chart_update.py

```diff
@@ -824,6 +835,57 @@ def _find_fiscal_positions(self, chart_template_ids, mapping_taxes,
                     'notes': notes,
                 })
 
+    @api.one
+    def _find_financial_reports(self):
+        wiz_fr = self.env['wizard.update.charts.accounts.financial.report']
+        # Remove previous financial reports
+        self.financial_report_ids.unlink()
+        # Search for new / updated accounts
+        root_account_id = self.chart_template_id.account_root_id.id
+        acc_templ_criteria = [
+            ('chart_template_id', '=', self.chart_template_id.id)]
+        if root_account_id:
+            acc_templ_criteria = ['|'] + acc_templ_criteria
+            acc_templ_criteria += [
+                '&', ('parent_id', 'child_of', [root_account_id]),
+                ('chart_template_id', '=', False)]
+        account_templates = self.env['account.account.template'].search(
+            acc_templ_criteria)
+        for account_template in account_templates:
+            template_fr_ids = set([fr.id for fr in
+                                   account_template.financial_report_ids])
+            # Ensure the account template is on the map (search for the mapped
+            # account id).
+            account = self.map_account_template(account_template)
```

Comment:
Instead of "ensuring" that is on the map, you should call always this method later, so you don't need to make this "pre-caching". In this case, I'm seeing that you don't use again the mapping dictionary, so it's enough removing the comment.